addons/web/static/src/core/dialog/dialog.scss

```diff
@@ -28,11 +28,11 @@
             margin: 0; // Reset boostrap.
         }
         
-        @include media-breakpoint-down(md) {
+        @include media-breakpoint-down(sm) {
```

Comment:
This is huge and it would require testing every modal in odoo... it's a bit out of scope for this PR 😅

Anyway, the aimed result it's debatable IMO. Makes sense for this modal specifically, but there are circumstances where you don't want a full-width button (eg. a single "OK" confirmation scenario)... and/or you want the vertical padding to adapt accordingly (?)... to be discussed :)

Reply:
Yes indeed it's huge and I expected you to say that, it was more to raise the concern 👍  So yeah, I agree it has to be discussed, at the moment the way these buttons wrap is really non optimal. Should we create a task in our pipeline ?

| Odd number of btns | Long labels |
|--------|--------|
| <img width="376" alt="Screenshot 2024-06-07 at 11 12 09" src="https://github.com/odoo-dev/odoo/assets/110090660/d9a480b5-3eb4-4fb3-9346-303e33b1c0ab"> | <img width="372" alt="Screenshot 2024-06-07 at 11 11 46" src="https://github.com/odoo-dev/odoo/assets/110090660/96e8c3d6-c3b3-4ef2-b6c8-d0a20aaecd7a"> |

Reply:
I don't think that's necessary 'cause the "_3 button with long labels_" it's non existent in the backend 😉